### PR TITLE
feat(eventing): add nexus events in protobuf

### PR DESCRIPTION
**Pull request by datacore-vvarakantham**
Added nexus category and rebuild, stateChange actions in event protobuf

### CODE REVIEW

File: events-api/protobuf/v1/event.proto [hunk 1]
```diff
@@ -23,13 +23,17 @@ enum EventCategory {
...
+  Nexus = 3;
 }
 
 // Event Action
 enum EventAction {
   UnknownAction = 0;
   Create = 1;
   Delete = 2;
+  StateChange = 3;
+  RebuildBegin = 4;
```
Comment (tiagolobocastro) [inline events-api/protobuf/v1/event.proto:35]: Are we simply interested in begin/end, or do we want to specify rebuild source/destination? If so, where would event specific data go into?

Comment (datacore-vvarakantham) [reply events-api/protobuf/v1/event.proto:35]: We can add data specific to particular event categories.

Comment (sinhaashish) [reply events-api/protobuf/v1/event.proto:35]: But why do the stats need source and destination. IMO source and destination of rebuilds are for logs, where we need to debug. But why to show that in reports?

Comment (tiagolobocastro) [reply events-api/protobuf/v1/event.proto:35]: Stats? I thought these were generic events and stats is just one use-case ?

Comment (sinhaashish) [reply events-api/protobuf/v1/event.proto:35]: hmm, for other use case it make sense. i though wrt to stats only.

Comment (datacore-vvarakantham) [reply events-api/protobuf/v1/event.proto:35]: We need source/destination for rebuilds. I will work on the event specific data in the future PRs.

**Review by chriswldenyer — COMMENTED** [review]
Looks OK.
Ideally we would like event-specific data fo the rebuild, but that can be in a later PR.

**Review by chriswldenyer — APPROVED** [review]

**Review by tiagolobocastro — APPROVED** [review]

**Pull request merged**